frontend/src/pages/modelServing/screens/projects/useModelDeploymentNotification.ts

```diff
@@ -0,0 +1,96 @@
+import * as React from 'react';
+import { useNavigate } from 'react-router-dom';
+import useNotification from '~/utilities/useNotification';
+import {
+  NotificationResponseStatus,
+  NotificationWatcherContext,
+} from '~/concepts/notificationWatcher/NotificationWatcherContext';
+import { getInferenceService } from '~/api';
+import { InferenceServiceModelState } from '~/pages/modelServing/screens/types';
+import {
+  getInferenceServiceLastFailureReason,
+  getInferenceServiceModelState,
+} from '~/pages/modelServing/screens/global/utils';
+import { useModelStatus } from '~/pages/modelServing/screens/global/useModelStatus';
+
+type ModelDeploymentNotification = {
+  watchDeployment: () => void;
+};
+
+export const useModelDeploymentNotification = (
+  namespace: string,
+  modelName: string,
+  isKserve: boolean,
+): ModelDeploymentNotification => {
+  const navigate = useNavigate();
+  const notification = useNotification();
+  const { registerNotification } = React.useContext(NotificationWatcherContext);
+  const [modelStatus] = useModelStatus(namespace, modelName, isKserve);
```

Comment:
@emilys314 hopefully we could remove the isKserve part here from useModelStatus during the refactor